src/Converter/JsonConverter.php

```diff
@@ -829,11 +829,12 @@ public function convertObject(array $data = null)
         if (isset($data[JSONConstants::JSON_OBJECT_SUCCINCT_PROPERTIES])) {
             $properties = $this->convertSuccinctProperties(
                 $data[JSONConstants::JSON_OBJECT_SUCCINCT_PROPERTIES] ?? [],
-                $data[JSONConstants::JSON_OBJECT_PROPERTIES_EXTENSION] ?? []);
+                (array) ($data[JSONConstants::JSON_OBJECT_PROPERTIES_EXTENSION] ?? [])
```

Comment:
no need to cast the empty array to an array. 
`(array) $data[JSONConstants::JSON_OBJECT_PROPERTIES_EXTENSION] ?? []`
should be sufficient

Reply:
Or am i missing anything?

Reply:
I think the thought here is that there is no guarantee that the value of `$data[JSONConstants::JSON_OBJECT_PROPERTIES_EXTENSION]` is an array as the method is public and $data is passed as an array.  That said, removing the cast is probably sufficient.  What is your preference?

Reply:
The problem is in the evaluation order:

`(array) $data[JSONConstants::JSON_OBJECT_PROPERTIES_EXTENSION]` gets evaluated first and causes the E_NOTICE already then - before the `??` even has a chance to evaluate. The parenthesis makes sure that evaluation happens first and then the output (which you're right may already be an array) gets cast. Removing the cast also works but as @marcelto says that would potentially pass through values that are not an array.

Reply:
Ok i got it...
But we should first check what really happens... @marcelto can you please describe when this issue occurs?
As far as i can see this value should by an array anyway. So if it is not something else is wrong. But not sure about that. Simply casting the value to an array could end in some unexplained behavior...

Reply:
The error is triggered when I call `createDocument` on `\Dkd\PhpCmis\Session`

	Undefined index: propertiesExtension
	 C:\repos\rcm-document-repo\vendor\dkd\php-cmis\src\Converter\JsonConverter.php:837
	 C:\repos\rcm-document-repo\vendor\dkd\php-cmis\src\Bindings\Browser\ObjectService.php:192
	 C:\repos\rcm-document-repo\vendor\dkd\php-cmis\src\Session.php:381
	 C:\repos\rcm-document-repo\src\Adapter\Cmis\CmisAdapter.php:222
	 C:\repos\rcm-document-repo\test\Cmis\CmisTest.php:80

Reply:
@marcelto what is the content of CmisTest.php?
Looks like you're passing invalid data into createDocument...

Reply:
Not exactly, it isn't an error, just a notice (my mistake).  It is causing my unit test to fail even though the document is created successfully.  You should be able to reproduce it with CreateDocument.php in the examples you provide.